In [3]:
import pandas as pd
from sklearn import preprocessing


In [4]:
df=pd.read_csv("train.csv")

In [5]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 889 entries, 0 to 888
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  889 non-null    int64  
 1   Survived     889 non-null    int64  
 2   Pclass       889 non-null    int64  
 3   Name         889 non-null    object 
 4   Sex          889 non-null    object 
 5   Age          889 non-null    float64
 6   SibSp        889 non-null    int64  
 7   Parch        889 non-null    int64  
 8   Ticket       889 non-null    object 
 9   Fare         889 non-null    float64
 10  Cabin        202 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.5+ KB


## *Encoding of categorical variables*

In [7]:
from sklearn.preprocessing import LabelEncoder
encoder=LabelEncoder()

In [8]:
df['Sex']=encoder.fit_transform(df['Sex'])
df['Embarked']=encoder.fit_transform(df['Embarked'])

In [9]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1      0   
2                             Heikkinen, Miss. Laina    0  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1      0   
4                           Allen, Mr. William Henry    1  35.0      0      0   

             Ticket     Fare Cabin  Embarked  
0         A/5 21171   7.2500   NaN         2  
1          PC 17599  71.2833   C85         0  
2  STON/O2. 3101282   7.9250   NaN         2  
3            113803  53.1000  C123         2  
4            373450   8.0500   NaN         2

## *Check for null values*

In [10]:
df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age              0
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         0
dtype: int64

## *Dropping of irrelevant features*

In [11]:
df=df.drop(['PassengerId','Cabin','Ticket','Name'],axis=1)
df.head()

Survived  Pclass  Sex   Age  SibSp  Parch     Fare  Embarked
0         0       3    1  22.0      1      0   7.2500         2
1         1       1    0  38.0      1      0  71.2833         0
2         1       3    0  26.0      0      0   7.9250         2
3         1       1    0  35.0      1      0  53.1000         2
4         0       3    1  35.0      0      0   8.0500         2

## *Splitting DataSet*

In [13]:
y=df['Pclass']
y.head()

0    3
1    1
2    3
3    1
4    3
Name: Pclass, dtype: int64

In [14]:
x=df.drop(['Pclass'],axis=1)
x.head()

Survived  Sex   Age  SibSp  Parch     Fare  Embarked
0         0    1  22.0      1      0   7.2500         2
1         1    0  38.0      1      0  71.2833         0
2         1    0  26.0      0      0   7.9250         2
3         1    0  35.0      1      0  53.1000         2
4         0    1  35.0      0      0   8.0500         2

In [15]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=0)
from sklearn import neighbors

In [16]:
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix
    # Train model
def KNN(n):
    for i in range(1,n):
        knn = neighbors.KNeighborsClassifier(n_neighbors=i)
        print("\nvalue of n_neighbors =",i)
        knn.fit(x_train, y_train)
        y_preds=knn.predict(x_test)
        print("\n Confusion Matrix:\n",confusion_matrix(y_test,y_preds))
        print("\n Accuracy Score \n",accuracy_score(y_test,y_preds))
        print("-----------------------------------------------------------------------------------------")

In [23]:
KNN(268)


value of n_neighbors = 1

 Confusion Matrix:
 [[ 62   6   2]
 [  3  37   9]
 [  1   8 139]]

 Accuracy Score 
 0.8913857677902621
-----------------------------------------------------------------------------------------

value of n_neighbors = 2

 Confusion Matrix:
 [[ 67   2   1]
 [  8  37   4]
 [  4  17 127]]

 Accuracy Score 
 0.8651685393258427
-----------------------------------------------------------------------------------------

value of n_neighbors = 3

 Confusion Matrix:
 [[ 63   5   2]
 [  7  28  14]
 [  3   5 140]]

 Accuracy Score 
 0.8651685393258427
-----------------------------------------------------------------------------------------

value of n_neighbors = 4

 Confusion Matrix:
 [[ 61   7   2]
 [  8  30  11]
 [  4  15 129]]

 Accuracy Score 
 0.8239700374531835
-----------------------------------------------------------------------------------------

value of n_neighbors = 5

 Confusion Matrix:
 [[ 58  10   2]
 [  7  28  14]
 [  3   9 136]]

 Accuracy Score 
 0.83


 Confusion Matrix:
 [[ 60   9   1]
 [  7  28  14]
 [  5  10 133]]

 Accuracy Score 
 0.8277153558052435
-----------------------------------------------------------------------------------------

value of n_neighbors = 14

 Confusion Matrix:
 [[ 60   8   2]
 [  8  27  14]
 [  5  11 132]]

 Accuracy Score 
 0.8202247191011236
-----------------------------------------------------------------------------------------

value of n_neighbors = 15

 Confusion Matrix:
 [[ 60   7   3]
 [  7  27  15]
 [  5  13 130]]

 Accuracy Score 
 0.8127340823970037
-----------------------------------------------------------------------------------------

value of n_neighbors = 16

 Confusion Matrix:
 [[ 60   7   3]
 [  7  28  14]
 [  5  13 130]]

 Accuracy Score 
 0.8164794007490637
-----------------------------------------------------------------------------------------

value of n_neighbors = 17

 Confusion Matrix:
 [[ 60   7   3]
 [  7  27  15]
 [  5  12 131]]

 Accuracy Score 
 0.8164794007490637
-------


 Confusion Matrix:
 [[ 61   6   3]
 [  7  27  15]
 [  8  14 126]]

 Accuracy Score 
 0.8014981273408239
-----------------------------------------------------------------------------------------

value of n_neighbors = 24

 Confusion Matrix:
 [[ 62   5   3]
 [  7  27  15]
 [  9  16 123]]

 Accuracy Score 
 0.7940074906367042
-----------------------------------------------------------------------------------------

value of n_neighbors = 25

 Confusion Matrix:
 [[ 62   6   2]
 [  7  26  16]
 [  9  14 125]]

 Accuracy Score 
 0.797752808988764
-----------------------------------------------------------------------------------------

value of n_neighbors = 26

 Confusion Matrix:
 [[ 61   5   4]
 [  7  25  17]
 [  9  15 124]]

 Accuracy Score 
 0.7865168539325843
-----------------------------------------------------------------------------------------

value of n_neighbors = 27

 Confusion Matrix:
 [[ 62   5   3]
 [  7  26  16]
 [  9  14 125]]

 Accuracy Score 
 0.797752808988764
---------


 Accuracy Score 
 0.7902621722846442
-----------------------------------------------------------------------------------------

value of n_neighbors = 34

 Confusion Matrix:
 [[ 62   5   3]
 [  7  24  18]
 [ 10  14 124]]

 Accuracy Score 
 0.7865168539325843
-----------------------------------------------------------------------------------------

value of n_neighbors = 35

 Confusion Matrix:
 [[ 62   5   3]
 [  7  22  20]
 [ 10  14 124]]

 Accuracy Score 
 0.7790262172284644
-----------------------------------------------------------------------------------------

value of n_neighbors = 36

 Confusion Matrix:
 [[ 62   5   3]
 [  7  23  19]
 [ 10  14 124]]

 Accuracy Score 
 0.7827715355805244
-----------------------------------------------------------------------------------------

value of n_neighbors = 37

 Confusion Matrix:
 [[ 62   5   3]
 [  7  22  20]
 [  9  13 126]]

 Accuracy Score 
 0.7865168539325843
--------------------------------------------------------------------------


 Confusion Matrix:
 [[ 61   5   4]
 [  7  19  23]
 [  7  12 129]]

 Accuracy Score 
 0.7827715355805244
-----------------------------------------------------------------------------------------

value of n_neighbors = 42

 Confusion Matrix:
 [[ 61   5   4]
 [  7  20  22]
 [  7  12 129]]

 Accuracy Score 
 0.7865168539325843
-----------------------------------------------------------------------------------------

value of n_neighbors = 43

 Confusion Matrix:
 [[ 61   6   3]
 [  7  19  23]
 [  7  12 129]]

 Accuracy Score 
 0.7827715355805244
-----------------------------------------------------------------------------------------

value of n_neighbors = 44

 Confusion Matrix:
 [[ 61   7   2]
 [  7  20  22]
 [  7  12 129]]

 Accuracy Score 
 0.7865168539325843
-----------------------------------------------------------------------------------------

value of n_neighbors = 45

 Confusion Matrix:
 [[ 61   7   2]
 [  7  19  23]
 [  7  12 129]]

 Accuracy Score 
 0.7827715355805244
-------


 Confusion Matrix:
 [[ 61   6   3]
 [  7  17  25]
 [  7  12 129]]

 Accuracy Score 
 0.7752808988764045
-----------------------------------------------------------------------------------------

value of n_neighbors = 49

 Confusion Matrix:
 [[ 62   6   2]
 [  7  16  26]
 [  7  12 129]]

 Accuracy Score 
 0.7752808988764045
-----------------------------------------------------------------------------------------

value of n_neighbors = 50

 Confusion Matrix:
 [[ 61   7   2]
 [  7  16  26]
 [  7  12 129]]

 Accuracy Score 
 0.7715355805243446
-----------------------------------------------------------------------------------------

value of n_neighbors = 51

 Confusion Matrix:
 [[ 61   6   3]
 [  7  14  28]
 [  5  12 131]]

 Accuracy Score 
 0.7715355805243446
-----------------------------------------------------------------------------------------

value of n_neighbors = 52

 Confusion Matrix:
 [[ 61   7   2]
 [  7  15  27]
 [  7  11 130]]

 Accuracy Score 
 0.7715355805243446
-------


 Confusion Matrix:
 [[ 62   5   3]
 [  7  12  30]
 [  5  10 133]]

 Accuracy Score 
 0.7752808988764045
-----------------------------------------------------------------------------------------

value of n_neighbors = 58

 Confusion Matrix:
 [[ 60   5   5]
 [  6  12  31]
 [  5  10 133]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 59

 Confusion Matrix:
 [[ 60   5   5]
 [  6  11  32]
 [  5  11 132]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 60

 Confusion Matrix:
 [[ 60   5   5]
 [  6  12  31]
 [  5  11 132]]

 Accuracy Score 
 0.7640449438202247
-----------------------------------------------------------------------------------------

value of n_neighbors = 61

 Confusion Matrix:
 [[ 60   5   5]
 [  6  11  32]
 [  5  11 132]]

 Accuracy Score 
 0.7602996254681648
-------


 Confusion Matrix:
 [[ 57   6   7]
 [  5  11  33]
 [  5   9 134]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 67

 Confusion Matrix:
 [[ 57   6   7]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 68

 Confusion Matrix:
 [[ 57   6   7]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 69

 Confusion Matrix:
 [[ 56   7   7]
 [  5  11  33]
 [  5   6 137]]

 Accuracy Score 
 0.7640449438202247
-----------------------------------------------------------------------------------------

value of n_neighbors = 70

 Confusion Matrix:
 [[ 57   6   7]
 [  5  11  33]
 [  5   7 136]]

 Accuracy Score 
 0.7640449438202247
-------


 Confusion Matrix:
 [[ 56   7   7]
 [  5  11  33]
 [  5   6 137]]

 Accuracy Score 
 0.7640449438202247
-----------------------------------------------------------------------------------------

value of n_neighbors = 76

 Confusion Matrix:
 [[ 56   7   7]
 [  5  11  33]
 [  5   5 138]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 77

 Confusion Matrix:
 [[ 57   6   7]
 [  5  11  33]
 [  5   5 138]]

 Accuracy Score 
 0.7715355805243446
-----------------------------------------------------------------------------------------

value of n_neighbors = 78

 Confusion Matrix:
 [[ 56   5   9]
 [  5  11  33]
 [  5   6 137]]

 Accuracy Score 
 0.7640449438202247
-----------------------------------------------------------------------------------------

value of n_neighbors = 79

 Confusion Matrix:
 [[ 56   5   9]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7565543071161048
-------


 Confusion Matrix:
 [[ 55   5  10]
 [  5  10  34]
 [  5   7 136]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 85

 Confusion Matrix:
 [[ 55   6   9]
 [  5  10  34]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 86

 Confusion Matrix:
 [[ 55   6   9]
 [  5  10  34]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 87

 Confusion Matrix:
 [[ 55   6   9]
 [  5  10  34]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 88

 Confusion Matrix:
 [[ 54   6  10]
 [  5  10  34]
 [  5   8 135]]

 Accuracy Score 
 0.7453183520599251
-------


 Confusion Matrix:
 [[ 54   5  11]
 [  5  10  34]
 [  5   6 137]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 95

 Confusion Matrix:
 [[ 54   6  10]
 [  5  10  34]
 [  5   6 137]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 96

 Confusion Matrix:
 [[ 55   4  11]
 [  5  10  34]
 [  5   6 137]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 97

 Confusion Matrix:
 [[ 53   6  11]
 [  5  10  34]
 [  5   7 136]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 98

 Confusion Matrix:
 [[ 54   6  10]
 [  5  10  34]
 [  5   5 138]]

 Accuracy Score 
 0.7565543071161048
-------


 Confusion Matrix:
 [[ 53   7  10]
 [  6  10  33]
 [  5  10 133]]

 Accuracy Score 
 0.7340823970037453
-----------------------------------------------------------------------------------------

value of n_neighbors = 105

 Confusion Matrix:
 [[ 53   7  10]
 [  6  11  32]
 [  5  10 133]]

 Accuracy Score 
 0.7378277153558053
-----------------------------------------------------------------------------------------

value of n_neighbors = 106

 Confusion Matrix:
 [[ 53   6  11]
 [  5  12  32]
 [  5  10 133]]

 Accuracy Score 
 0.7415730337078652
-----------------------------------------------------------------------------------------

value of n_neighbors = 107

 Confusion Matrix:
 [[ 53   9   8]
 [  5  12  32]
 [  5  10 133]]

 Accuracy Score 
 0.7415730337078652
-----------------------------------------------------------------------------------------

value of n_neighbors = 108

 Confusion Matrix:
 [[ 53   8   9]
 [  5  12  32]
 [  5   9 134]]

 Accuracy Score 
 0.7453183520599251
---


 Confusion Matrix:
 [[ 53   8   9]
 [  5  12  32]
 [  5   9 134]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 113

 Confusion Matrix:
 [[ 53   8   9]
 [  5  12  32]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 114

 Confusion Matrix:
 [[ 53   9   8]
 [  5  12  32]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 115

 Confusion Matrix:
 [[ 53  11   6]
 [  5  12  32]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 116

 Confusion Matrix:
 [[ 53  11   6]
 [  5  12  32]
 [  5   8 135]]

 Accuracy Score 
 0.7490636704119851
---


 Confusion Matrix:
 [[ 53  12   5]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 123

 Confusion Matrix:
 [[ 53  12   5]
 [  5  10  34]
 [  5   8 135]]

 Accuracy Score 
 0.7415730337078652
-----------------------------------------------------------------------------------------

value of n_neighbors = 124

 Confusion Matrix:
 [[ 53  13   4]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 125

 Confusion Matrix:
 [[ 53  12   5]
 [  5  11  33]
 [  5   8 135]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 126

 Confusion Matrix:
 [[ 53  12   5]
 [  5  11  33]
 [  5   7 136]]

 Accuracy Score 
 0.7490636704119851
---


 Confusion Matrix:
 [[ 53  12   5]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 132

 Confusion Matrix:
 [[ 53  13   4]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 133

 Confusion Matrix:
 [[ 53  13   4]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 134

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 135

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   3 140]]

 Accuracy Score 
 0.7602996254681648
---


 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   3 140]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 141

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   3 140]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 142

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 143

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 144

 Confusion Matrix:
 [[ 52  13   5]
 [  5  11  33]
 [  5   4 139]]

 Accuracy Score 
 0.7565543071161048
---


 Confusion Matrix:
 [[ 52  11   7]
 [  5  11  33]
 [  5   1 142]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 149

 Confusion Matrix:
 [[ 52  11   7]
 [  5  11  33]
 [  5   1 142]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 150

 Confusion Matrix:
 [[ 52  11   7]
 [  5  11  33]
 [  5   1 142]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 151

 Confusion Matrix:
 [[ 52  11   7]
 [  5  11  33]
 [  5   1 142]]

 Accuracy Score 
 0.7677902621722846
-----------------------------------------------------------------------------------------

value of n_neighbors = 152

 Confusion Matrix:
 [[ 52  11   7]
 [  5  10  34]
 [  5   1 142]]

 Accuracy Score 
 0.7640449438202247
---


 Confusion Matrix:
 [[ 52   9   9]
 [  5   9  35]
 [  5   1 142]]

 Accuracy Score 
 0.7602996254681648
-----------------------------------------------------------------------------------------

value of n_neighbors = 157

 Confusion Matrix:
 [[ 52   9   9]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 158

 Confusion Matrix:
 [[ 52   9   9]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 159

 Confusion Matrix:
 [[ 52   9   9]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 160

 Confusion Matrix:
 [[ 52   9   9]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
---


 Confusion Matrix:
 [[ 52   9   9]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 164

 Confusion Matrix:
 [[ 52   9   9]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 165

 Confusion Matrix:
 [[ 52   9   9]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 166

 Confusion Matrix:
 [[ 52   8  10]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 167

 Confusion Matrix:
 [[ 52   7  11]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
---


 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 172

 Confusion Matrix:
 [[ 52   6  12]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7528089887640449
-----------------------------------------------------------------------------------------

value of n_neighbors = 173

 Confusion Matrix:
 [[ 52   6  12]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 174

 Confusion Matrix:
 [[ 50   8  12]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 175

 Confusion Matrix:
 [[ 52   6  12]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
----------------------------------------------------------------------


 Confusion Matrix:
 [[ 52   6  12]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 180

 Confusion Matrix:
 [[ 50   7  13]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 181

 Confusion Matrix:
 [[ 52   5  13]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 182

 Confusion Matrix:
 [[ 52   5  13]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
-----------------------------------------------------------------------------------------

value of n_neighbors = 183

 Confusion Matrix:
 [[ 52   5  13]
 [  5   7  37]
 [  5   0 143]]

 Accuracy Score 
 0.7565543071161048
---


 Confusion Matrix:
 [[ 51   5  14]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 188

 Confusion Matrix:
 [[ 51   5  14]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7490636704119851
-----------------------------------------------------------------------------------------

value of n_neighbors = 189

 Confusion Matrix:
 [[ 50   5  15]
 [  5   6  38]
 [  5   0 143]]

 Accuracy Score 
 0.7453183520599251
-----------------------------------------------------------------------------------------

value of n_neighbors = 190

 Confusion Matrix:
 [[ 50   5  15]
 [  5   5  39]
 [  5   0 143]]

 Accuracy Score 
 0.7415730337078652
-----------------------------------------------------------------------------------------

value of n_neighbors = 191

 Confusion Matrix:
 [[ 50   5  15]
 [  5   5  39]
 [  5   0 143]]

 Accuracy Score 
 0.7415730337078652
---


 Confusion Matrix:
 [[ 48   7  15]
 [  5   2  42]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 196

 Confusion Matrix:
 [[ 48   7  15]
 [  5   2  42]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 197

 Confusion Matrix:
 [[ 48   7  15]
 [  5   2  42]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 198

 Confusion Matrix:
 [[ 48   7  15]
 [  5   2  42]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 199

 Confusion Matrix:
 [[ 48   7  15]
 [  5   1  43]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
---


 Confusion Matrix:
 [[ 48   6  16]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 203

 Confusion Matrix:
 [[ 48   6  16]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 204

 Confusion Matrix:
 [[ 48   6  16]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 205

 Confusion Matrix:
 [[ 48   6  16]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 206

 Confusion Matrix:
 [[ 48   6  16]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
---


 Confusion Matrix:
 [[ 48   5  17]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 211

 Confusion Matrix:
 [[ 48   5  17]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 212

 Confusion Matrix:
 [[ 48   4  18]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 213

 Confusion Matrix:
 [[ 48   4  18]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 214

 Confusion Matrix:
 [[ 48   4  18]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
---


 Confusion Matrix:
 [[ 48   4  18]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 219

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 220

 Confusion Matrix:
 [[ 48   3  19]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 221

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 222

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
---


 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 227

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 228

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 229

 Confusion Matrix:
 [[ 48   2  20]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 230

 Confusion Matrix:
 [[ 48   1  21]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
---


 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 235

 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 236

 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 237

 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 238

 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
---


 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 243

 Confusion Matrix:
 [[ 48   0  22]
 [  4   2  43]
 [  5   0 143]]

 Accuracy Score 
 0.7228464419475655
-----------------------------------------------------------------------------------------

value of n_neighbors = 244



 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 245

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 246

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 247

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 248

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
---

-----------------------------------------------------------------------------------------

value of n_neighbors = 251

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 252

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 253

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 254

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighb


 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 259

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 260

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 261

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 262

 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
---


 Confusion Matrix:
 [[ 48   0  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7191011235955056
-----------------------------------------------------------------------------------------

value of n_neighbors = 266

 Confusion Matrix:
 [[ 47   0  23]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7153558052434457
-----------------------------------------------------------------------------------------

value of n_neighbors = 267

 Confusion Matrix:
 [[ 47   1  22]
 [  4   1  44]
 [  5   0 143]]

 Accuracy Score 
 0.7153558052434457
-----------------------------------------------------------------------------------------


## *Conclusion*

**From above it is clear that accuracy score is high 89%, when n_neighbors value is 1 and out of 267 records 238 records are correctly classified**